packages/core/src/dynamoDb/utils/dynamodb.ts

```diff
@@ -216,3 +217,36 @@ function getAttributeValue(attribute: AttributeValue): { key: string; value: any
     }
     return undefined
 }
+
+export async function deleteItem(
+    tableName: string,
+    selectedRow: RowData,
+    tableSchema: TableSchema,
+    regionCode: string,
+    client = new DynamoDbClient(regionCode)
+) {
+    const partitionKeyName = tableSchema.partitionKey.name
+    const partitionKeyValue = selectedRow[partitionKeyName]
+
+    const deleteRequest: DynamoDB.DocumentClient.DeleteItemInput = {
+        TableName: tableName,
+        Key: {
+            partitionKeyName: {
+                S: partitionKeyValue,
+            } as any,
+        },
+    }
+    if (tableSchema.sortKey) {
+        const sortKeyName = tableSchema.sortKey.name
+        const sortKeyValue = selectedRow[sortKeyName]
+        deleteRequest.Key[sortKeyName] = { S: sortKeyValue } as any
+    }
+
+    try {
+        const response = await client.deleteItem(deleteRequest)
```

Comment:
I think you could just `return client.deleteItem(deleteRequest)` instead of reassigning. Or alternativelty `return await client.deleteItem(deleteRequest)`.

https://stackoverflow.com/questions/38708550/difference-between-return-await-promise-and-return-promise has more info on the differences